server/src/internalClusterTest/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java

```diff
@@ -468,8 +469,15 @@ public void testMasterFailOverWithQueuedDeletes() throws Exception {
         unblockNode(repoName, dataNode);
         unblockNode(repoName, dataNode2);
 
-        assertAcked(firstDeleteFuture.get());
```

Comment:
Most of the time these go through just fine because we retry the delete requests one node disconnect and then they get batched up with the already running deletes from the previous master and all is well.
But if we fail over twice then the second fail-over will resolve the listeners with exceptions (this is on purpose, we don't want to be retrying (hence the exception wrapping) in this case yet again because we then get a mix of snapshot not found exception successful execution depending on the timing).